### ISSUE TITLE
Threadpool's queue configuration in parallel application context refresh.

### ISSUE THREAD

**Issue by alaneuler** [sofastack/sofa-boot#865]
Synchronous queue has its internal defect when using along side with CallerRunPolicy.

Considering following scenario:
1. all threads of `core` size are busy refreshing context
2. new refreshing task comes while it is time-consuming
3. main thread processes the time-consuming task by itself
4. threads in the thread pool finish their tasks but main thread gets no time submit new task

Threads in the pool is wasting time in waiting new tasks, which reduces overall parallelism.
